src/rdkafka_cgrp.c

```diff
@@ -992,6 +1022,14 @@ static void rd_kafka_cgrp_handle_JoinGroup (rd_kafka_t *rk,
                 goto err;
         }
 
+        if (rkcg->rkcg_assignor && rkcg->rkcg_assignor != rkas) {
```

Comment:
Should assignor state be scrapped if the subscription is changed through a new subscribe/unsubscribe() call?

Reply:
i don't think so - it's still relevant. maybe the new subscription overlaps, in which case the old assignment is useful information to the assignor.

also, the java consumer doesn't do this afaict.